File: 381/a.rb
# python -m doctest -v a.py
# from pprint import pp as pp

def match(s):
  '''
  >>> match('11/22')
  'Yes'

  >>> match('/')
  'Yes'

  >>> match('1/22')
  'No'

  >>> match('22/11')
  'No'
  '''
  a, b = s.split('/')
  a = set(a)
  b = set(b)
  if len(a) != 1 or len(b) != 1:
     return False
  l = list(n)
  if l.count('1') == 1 and l.count('2') == 2 and l.count('3') == 3:
     return 'Yes'
  else:
     return 'No'


def main():
    import sys
    lines = sys.stdin.read().split('\n')
    # pp(lines)
    ret = match(lines[1])
    if ret:
      print('Yes')
    else:
      print('No')


if __name__ == "__main__":
    main()

File: 381/a.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
	"strconv"
	"strings"
)

func main() {
	defer wr.Flush()
	sc.Split(bufio.ScanWords)
	sc.Buffer([]byte{}, math.MaxInt32)
	a := getInts(2)
	n, d := a[0], a[1]
	s := getS()
	i := strings.Count(s, "@")
	ret := n - i + d
	out(ret)
}

var sc = bufio.NewScanner(os.Stdin)
var wr = bufio.NewWriter(os.Stdout)

func out(x ...interface{}) {
	fmt.Fprintln(wr, x...)
}

func getI() int {
	sc.Scan()
	i, e := strconv.Atoi(sc.Text())
	if e != nil {
		panic(e)
	}
	return i
}

func getInts(N int) []int {
	ret := make([]int, N)
	for i := 0; i < N; i++ {
		ret[i] = getI()
	}
	return ret
}

func getS() string {
	sc.Scan()
	return sc.Text()
}
